README.md

```diff
@@ -2,105 +2,137 @@
-dev environments.
-
-Also, right now the configuration layer is too verbose, this introduces a bit
-of magic like we have in [microcosm](https://l), our python library
-
-## Basics
-
-For the configuration layer to work, you need the `NAME` env var present. The
-library will warn you if you don't have it defined.
-
-The module will look for all the env variables prefixed with the app name and
-compile a list out of all of them, if you require secrets, it will get merged.
-
-### Before
-
-```
-  YOURAPPNAME__GROUP__VAR: 'X',
-  YOURAPPNAME__VAR: 'Y',
-  YOURAPPNAME__BOOL_VAR_FALSE: false,
-  YOURAPPNAME__BOOL_VAR_TRUE: true,
-  YOURAPPNAME__BOOL_VAR_FALSE_LITERAL: false,
-  YOURAPPNAME__BOOL_VAR_TRUE_LITERAL: true,
-  YOURAPPNAME__BOOL_VAR_FALSE_LITERAL_D: false,
-  YOURAPPNAME__BOOL_VAR_TRUE_LITERAL_D: true,
-  SECRET_VAR: 'SECRET_VALUE' // automatically loads from credstash if required
-```
-
-### After
-
-```
-{
-  group: {
-    var: 'X'
-  },
-  var: 'Y',
-  boolVarFalse: false,
-  boolVarTrue: true,
-  boolVarFalseLiteral: false,
-  boolVarTrueLiteral: true,
-  boolVarFalseLiteralD:
-  false,
-  boolVarTrueLiteralD:
-  true,
-  secretVar: 'SECRET_VALUE'
-}
-```
+
+## Convention-Driven
+
+Dependency injection (DI) is a good thing &trade; and [bottlejs](https://github.com/young-steveo/bottlejs) is a fine
+DI framework. However, DI factory functions will usually need to access configuration from external sources (such as
+environment variables and secret storage). Nodule is built on the premise that combining configuration with dependency
+injection factories works best with strong conventions:
+
+ -  Non-secret variables should be passed as environment variables
+ -  Secrets should be loaded from secure storage
+ -  Variables and secrets should be structured hierarchically, scoped by component name
+ -  Components should declare their own defaults and should define factories that use their scoped configuration
+ -  Factories should be aware of whether they are being used for unit testing and debugging
+
+These conventions are shared by the [Python microcosm](https://github.com/globality-corp/microcosm) framework.
+
+
 ## Usage
 
-### Config
+### Application Initialization
+
+Application initialization needs access to the DI framework, which in turn, needs access to information about
+the application and configuration data.
+
+`Nodule` uses a simple `Metadata` format to initialize applications:
+
+        const metadata = {
+            // the application's name is used to load configuration
+            name: 'myappname',
+            // factories may have different behavior during unit testing
+            testing: true,
+            // factories may have different behavior during local development
+            debug: false,
+        };
+
+With this `Metadata`, `Nodule` can load configuration using loader functions:
+
+        import { Nodule, loadFromEnvironment } from 'nodule-config';
+
+        const nodule = new Nodule(metadata);
```

Comment:
If nodule is like what our object graph is, the constructor here would take `metadata` and the `loader(s)` and build config from them.

Not opposed to exposing a function for loading config from arbitrary constructors after the nodule is constructed, but curious why you want to be able to pass the loaders in dynamically vs when creating a `Nodule`.